pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/rebalance/TableRebalancer.java

```diff
@@ -559,6 +588,147 @@ private RebalanceResult doRebalance(TableConfig tableConfig, RebalanceConfig reb
+            RebalanceSummaryResult.ServerStatus.REMOVED, 0, entry.getValue().size(), 0, entry.getValue().size(), 0,
+            instanceToTagsMap.getOrDefault(server, null)));
+      }
+    }
+
+    RebalanceSummaryResult.RebalanceChangeInfo numSegmentsInSingleReplica
+        = new RebalanceSummaryResult.RebalanceChangeInfo(currentAssignment.size(), targetAssignment.size());
+
+    int existingNumberSegmentsTotal = existingReplicationFactor * currentAssignment.size();
+    int newNumberSegmentsTotal = newReplicationFactor * targetAssignment.size();
+    RebalanceSummaryResult.RebalanceChangeInfo numSegmentsAcrossAllReplicas
+        = new RebalanceSummaryResult.RebalanceChangeInfo(existingNumberSegmentsTotal, newNumberSegmentsTotal);
+
+    int totalSegmentsToBeMoved = newNumberSegmentsTotal - segmentsNotMoved;
+
+    long tableSizePerReplicaInBytes = calculateTableSizePerReplicaInBytes(tableNameWithType);
+    long averageSegmentSizeInBytes = tableSizePerReplicaInBytes <= 0 ? tableSizePerReplicaInBytes
+        : tableSizePerReplicaInBytes / ((long) currentAssignment.size());
+    long totalEstimatedDataToBeMovedInBytes = tableSizePerReplicaInBytes <= 0 ? tableSizePerReplicaInBytes
+        : ((long) totalSegmentsToBeMoved) * averageSegmentSizeInBytes;
+    double estimatedTimeToRebalanceInSec = getEstimatedTimeToRebalanceInSec(totalEstimatedDataToBeMovedInBytes,
+        numServersGettingDataAdded);
+
+    RebalanceSummaryResult.ServerInfo serverInfo = new RebalanceSummaryResult.ServerInfo(numServersGettingDataAdded,
+        numServers, serverSegmentChangeInfoMap);
+    RebalanceSummaryResult.SegmentInfo segmentInfo = new RebalanceSummaryResult.SegmentInfo(totalSegmentsToBeMoved,
+        averageSegmentSizeInBytes, totalEstimatedDataToBeMovedInBytes, estimatedTimeToRebalanceInSec,
+        replicationFactor, numSegmentsInSingleReplica, numSegmentsAcrossAllReplicas);
+
+    LOGGER.info("Calculated rebalance summary for table: {} with rebalanceJobId: {}", tableNameWithType,
+        rebalanceJobId);
+    return new RebalanceSummaryResult(serverInfo, segmentInfo);
+  }
+
+  private static double getEstimatedTimeToRebalanceInSec(long totalEstimatedDataToBeMovedInBytes,
+      int numServersGettingDataAdded) {
+    double estimatedTimeToRebalanceInSec = totalEstimatedDataToBeMovedInBytes == 0 ? 0.0 : -1.0;
+    if (totalEstimatedDataToBeMovedInBytes > 0 && numServersGettingDataAdded > 0) {
+      // Do some processing to figure out what the time to download might be
+      // Assume that data is evenly distributed across all servers
+      long totalDataPerServerInBytes = totalEstimatedDataToBeMovedInBytes / numServersGettingDataAdded;
+      // TODO: Pick a good threshold to calculate estimated time to rebalance. For now assume 100 MB/s data download
+      //       + process rate
+      estimatedTimeToRebalanceInSec = ((double) totalDataPerServerInBytes) / (100.0D * 1024.0D * 1024.0D);
```

Comment:
I think maybe we skip this field, as it's not easy to estimate this accurately because the cluster setup can vary a lot. The summary contains the total amount of segments and data bytes a server may get now, so the user can estimate the duration for their envs accordingly.

Reply:
yeah, this is a tricky field for sure. I've removed it for now, but added a TODO for adding it back once we have a better understanding of a good estimate. This field was an ask for this summary PR.